### ISSUE TITLE
CoinGecko stopped working

### ISSUE THREAD

**Issue by nhct0** [hwki/SimpleBitcoinWidget#128]
CoinGecko no longer works for any coin, incl. BTC and ETH - seeing stale prices, red triangle with exclamation point icons.

For some fairly well-known coins, e.g., RPL (Rocket Pool), Nexo, CoinGecko happens to be the only available price source on the app, although nowadays they are actually listed on many app-included exchanges.

Noticed that this tracks with also just opened issue #127 (Kaspa).

v. 8.4.1, confirmed on two different Android phones.  Thanks.

**Comment by hwki** [comment]
Can you try recreating one of the widgets to see if it works? I think due to a prior bug fix any of the "partially supported" coins that work only on Coingecko might require being recreated.

**Comment by hwki** [comment]
I'm unable to reproduce, everything works fine for me. What country are you in?
I do see that CoinGecko has restricted calls to 30 per minute. Do you maybe have a lot of widgets?

**Comment by hwki** [comment]
@nhct0 The app has "full" support for a number of coins, where it knows about them and can pull them down from multiple exchanges. The rest are "partially" supported, where the list is pulled dynamically from coingecko. There are just too many crypto coins to add them all to the app.

For Rocket Pool, it is not supported on Kraken or Coinbase. The only exchange I integrate with that it is listed on aside from Coingecko is Binance: https://www.coingecko.com/en/coins/rocket-pool

That being said, I can look into adding it as well as support for some of the other exchanges listed there.
I can also add full support for Kaspa as well.

Regarding this issue though, I can't reproduce it for the life of me. Does it still happen if you do a fresh install of the app? If so is there any other configuration set up aside from the default? Do you see a value if you try to hit the API directly? (Just paste this url in a browser)
https://api.coingecko.com/api/v3/simple/price?ids=rocket-pool&vs_currencies=usd

**Comment by hwki** [comment]
Thanks. I have received reports from other users showing the same thing, blocked by the API. 

I pushed out an update yesterday that should play a little bit nicer with CoinGecko. Might help avoid getting blocked but not sure. I did add RocketPool and Nexo, so I would recommend switching them over to another exchange for now.

Leaving this issue open as I'm sure this isn't the last time this will be a problem. I might have to remove CoinGecko altogether in the future.
